### PR TITLE
volume mount sandbox should protect against path traversal

### PR DESCRIPTION
**Pull request by tgross**
The volume mounts for the LXC driver check that the path is not absolute, but
a relative path can be passed as the source and that can be used to escape the
task directory sandbox in the case where `volumes_enabled` is false (not the
default).

This issue was discovered while investigating https://github.com/hashicorp/nomad/issues/9303

### CODE REVIEW

**Comment by tgross** [comment]
While this works in local development, it looks like TravisCI is broken with the current `make deps`; this is because some of our dependencies have switched to go.mod.

File: lxc/lxc_test.go [hunk 1]
```diff
@@ -67,6 +67,34 @@ func TestLXCDriver_Mounts(t *testing.T) {
...
+	d := NewLXCDriver(testlog.HCLogger(t)).(*Driver)
+	d.config.Enabled = true
+	d.config.AllowVolumes = false
+
+	_, err := d.mountEntries(task, taskConfig)
+	require.EqualError(t, err, "absolute bind-mount volume in config but volumes are disabled")
+
+	taskConfig.Volumes = []string{"relative/../../../path2:usr-config/container/relative"}
+	_, err = d.mountEntries(task, taskConfig)
+	require.EqualError(t, err, "bind-mount path escapes task directory but volumes are disabled")
```
Comment (notnoop) [inline lxc/lxc_test.go:95]: Do we have a test for the positive case where a relative path is inside the task dir?

**Review by notnoop — APPROVED** [review]

File: lxc/lxc_test.go (hunk 1 above)
Comment (tgross) [reply lxc/lxc_test.go:95]: `TestLXCDriver_Mounts` should cover that case.

**Review by picatz — APPROVED** [review]

**Comment by tgross** [comment]
Working on getting the build fixed in https://github.com/hashicorp/nomad-driver-lxc/pull/22

**Pull request merged**